packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -834,6 +854,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
     if (widget.controller != oldWidget.controller) {
       oldWidget.controller.removeListener(_didChangeTextEditingValue);
       widget.controller.addListener(_didChangeTextEditingValue);
+      _hideSelectionOverlayIfNeeded();
```

Comment:
Selection will left over in a broken state if controller changed

Reply:
Is there a test covering this case?

Reply:
No, I will add one

Reply:
refactored. it turns out we should try to update the selection overlay instead of just hiding it